### DISCUSSION TITLE
Read parameters from car

### DISCUSSION THREAD

**Discussion by batal131** [jilleb/MQB-FPA#18] [Q&A] [answered]
Hey, Sorry for nooby question :)

How to read parameters from my car? Any easy method?

I have ODIS-E and VCDS.

**Comment by andy0172** [comment, marked as answer]
You need Odis Online connection and restore Dataset and Coding. Then you can grab the file from filesystem

**Reply by batal131** [reply]
Got it. Thanks.

**Reply by andy0172** [reply]
Or send me you VIN and I will do it for you.

**Reply by andy0172** [reply]
This is old GW and wont work with this tool here, so you wont need to grad it.
You need to switch to 3Q0 Gateway

**Reply by tntrio** [reply]
hm ok, thank you

**Reply by batal131** [reply]
> Or send me you VIN and I will do it for you.

How can I reach you? email?

**Reply by ivelliosy** [reply]
Hi @andy0172 , i think i messed up my FPA, now there is no profile showing in MMI. I appreciate if you hand me to restore my original dataset for gateway. My e-mail is cihanozen (@) gmail (dot) com

Audi A3
Part No SW: 3Q0 907 530 AC    HW: 3Q0 907 530 C
VIN : -CENSORED-

Thanks!

**Reply by Vlad0210** [reply]
Hello. I'm trying to read the parameter using Odis Online, but I can not find the parameter in the file system. Please tell me where is the parameters?

**Reply by komoornik** [reply]
> You need Odis Online connection and restore Dataset and Coding. Then you can grab the file from filesystem

There's some really useful info here in the wiki, but the important things are missing - any hints on how to export the data from gateway and then import it back? Like a simple step by step instructions.

You guys mention here it's not possible via VCP but in the script there's a VCP format that can be used.

**Reply by jilleb** [reply]
You cannot dump any data from the gateway.
Importing it is a matter of parametrization via VCP orr Odis.

**Reply by komoornik** [reply]
> You cannot dump any data from the gateway. Importing it is a matter of parametrization via VCP orr Odis.

Anyone willing to help me get the dataset for my VIN? :)

I have someone with VCP so the parametrization part can be done.

**Reply by batal131** [reply]
> > You cannot dump any data from the gateway. Importing it is a matter of parametrization via VCP orr Odis.
> 
> Anyone willing to help me get the dataset for my VIN? :)
> 
> I have someone with VCP so the parametrization part can be done.

Share your version. Maybe someone alreade have it.

**Reply by komoornik** [reply]
> > > You cannot dump any data from the gateway. Importing it is a matter of parametrization via VCP orr Odis.
> > 
> > 
> > Anyone willing to help me get the dataset for my VIN? :)
> > I have someone with VCP so the parametrization part can be done.
> 
> Share your version. Maybe someone alreade have it.

What exactly is needed?

The car is a Skoda Superb prod year 2021 (probably MY22?)

TMBAN7NP9M7078360

What I would like to achieve here is for the driving mode not to reset after ignition is off (including the setting for engine/gearbox).

**Reply by batal131** [reply]
> > > > You cannot dump any data from the gateway. Importing it is a matter of parametrization via VCP orr Odis.
> > > 
> > > 
> > > Anyone willing to help me get the dataset for my VIN? :)
> > > I have someone with VCP so the parametrization part can be done.
> > 
> > 
> > Share your version. Maybe someone alreade have it.
> 
> What exactly is needed?
> 
> The car is a Skoda Superb prod year 2021 (probably MY22?)
> 
> TMBAN7NP9M7078360
> 
> What I would like to achieve here is for the driving mode not to reset after ignition is off (including the setting for engine/gearbox).

Dataset version. 
Its possible to read dataset version without odis online. Almost any tool can do it.

**Reply by komoornik** [reply]
I'm still a bit newbie in those areas. I have OBD11. Where exactly I can find it?

Data from my gateway:

Hardware number: 3Q0907530C
Hardware version: 521
Software number: 3Q0907530AG
Software version: 5423
ODX version: 013026

**Reply by komoornik** [reply]
Anyone? ;)

**Reply by Carmaffia** [reply]
This is the output of VCDS: 

Address 19: CAN Gateway (J533)       Labels:| 3Q0-907-530-V1.clb
   Part No SW: 3Q0 907 530 M    HW: 3Q0 907 530 D
   Component: GW MQB EV     618 6322  
   ASAM Dataset: EV_GatewNF 013023
   ROD: EV_GatewNF_VW37.rod
   VCID: 392F2DF919DA6C13A0F-806C

**Reply by komoornik** [reply]
So it's ASAM Dataset: EV_GatewNF 013023?

I guess mine is EV_GatewNF 013026

**Reply by Carmaffia** [reply]
You could be right. So start searching from there.

**Reply by Krosta8** [reply]
You also have both different GW. Index D is MQB EV and index C is MQB classic. So different ASAM is not the problem I would say

**Reply by komoornik** [reply]
So here's my GW:
https://i.ibb.co/FszckCB/golf-gw.jpg

Anyone has dataset for it?

**Reply by komoornik** [reply]
I cannot sleep because of this ;)

Now that I know there is a solution for me to not click this damn mode button each time I start the engine.

Anyone who can grab the dataset from ODIS maybe?

**Reply by batal131** [reply]
Unfortunately I don't know where it is in OBD11, but I know that it is possible to get FPA version with it.
Here is how it should look in OBD11 (sorry, its in russian)
https://ibb.co/486mqPX

l4 - FPA version

**Reply by komoornik** [reply]
Ok, so my version would be 4B.

The data I have in Advanced identification is:
MY
22
4B
84

**Reply by komoornik** [reply]
Anyone have that kind of dataset? ;)

**Reply by komoornik** [reply]
@andy0172 are you able to help me with getting dataset for my car?

**Reply by komoornik** [reply]
Nobody can help with the dataset for my car? :(

**Reply by jjff2** [reply]
> Nobody can help with the dataset for my car? :(

if you have odis. you can grab it yourself.

**Reply by komoornik** [reply]
I do not have ODIS :(

ODIS online connection is needed?

Any tips or guide how to do that? I could find someone with ODIS.

**Reply by jjff2** [reply]
> I do not have ODIS :(
> 
> ODIS online connection is needed?
> 
> Any tips or guide how to do that? I could find someone with ODIS.

Odis online is needed.
Use special function. Retrieve your gateway’s data.

**Reply by komoornik** [reply]
My gateway is 3Q0907530C and from a sample xml file for this module I do not have START_ADDRESS="0x0000B80"

So where I could find FPA data? Sample file: https://pastebin.com/CesQFKPj

**Reply by cmaukr** [reply]
Hello everyone, tell me in which folder the dataset is located ib ODIS after online? Or am I doing something wrong?
Maybe someone has the original dataset 3q0907530aa

**Reply by G1Apples** [reply]
> My gateway is 3Q0907530C and from a sample xml file for this module I do not have START_ADDRESS="0x0000B80"
> 
> So where I could find FPA data? Sample file: https://pastebin.com/CesQFKPj

So have you got it done ?
Im ready to buy a VCP set and im also ready to learn all of this to get it done, but i fail at the first step of getting the data via odis online as i dont have acces to it.

**Reply by komoornik** [reply]
> > My gateway is 3Q0907530C and from a sample xml file for this module I do not have START_ADDRESS="0x0000B80"
> > So where I could find FPA data? Sample file: https://pastebin.com/CesQFKPj
> 
> So have you got it done ? Im ready to buy a VCP set and im also ready to learn all of this to get it done, but i fail at the first step of getting the data via odis online as i dont have acces to it.

I won't be able to help much i guess - I found a guy with ODIS online access and he downloaded the xml file for me.

The one from ODIS had the correct START_ADDRESS.

I have used the templates and he successfully "flashed" the modified file for me - it is working as intended:

https://www.youtube.com/shorts/A5Ge0qKtHto

**Reply by G1Apples** [reply]
> > > My gateway is 3Q0907530C and from a sample xml file for this module I do not have START_ADDRESS="0x0000B80"
> > > So where I could find FPA data? Sample file: https://pastebin.com/CesQFKPj
> > 
> > 
> > So have you got it done ? Im ready to buy a VCP set and im also ready to learn all of this to get it done, but i fail at the first step of getting the data via odis online as i dont have acces to it.
> 
> I won't be able to help much i guess - I found a guy with ODIS online access and he downloaded the xml file for me.
> 
> The one from ODIS had the correct START_ADDRESS.
> 
> I have used the templates and he successfully "flashed" the modified file for me - it is working as intended:
> 
> https://www.youtube.com/shorts/A5Ge0qKtHto

Is there a possibility that you can send me the contact or anyone else that could help me with the file via odis online? In any way thank you, that also helped me to get a bit further  (I just put my e-mail in here so you dont set him public Maxi3ds@yahoo.de)

**Reply by komoornik** [reply]
> > > > My gateway is 3Q0907530C and from a sample xml file for this module I do not have START_ADDRESS="0x0000B80"
> > > > So where I could find FPA data? Sample file: https://pastebin.com/CesQFKPj
> > > 
> > > 
> > > So have you got it done ? Im ready to buy a VCP set and im also ready to learn all of this to get it done, but i fail at the first step of getting the data via odis online as i dont have acces to it.
> > 
> > 
> > I won't be able to help much i guess - I found a guy with ODIS online access and he downloaded the xml file for me.
> > The one from ODIS had the correct START_ADDRESS.
> > I have used the templates and he successfully "flashed" the modified file for me - it is working as intended:
> > https://www.youtube.com/shorts/A5Ge0qKtHto
> 
> Is there a possibility that you can send me the contact or anyone else that could help me with the file via odis online? In any way thank you, that also helped me to get a bit further (I just put my e-mail in here so you dont set him public [Maxi3ds@yahoo.de](mailto:Maxi3ds@yahoo.de))

I do think the only way to get the file is the connect with ODIS to the car - so it needs to be direct access to the car.

Anyway, the guy has a business site:
https://retrofitlab.pl/

**Comment by NetCatx** [comment]
Im facing  the same Nobody question 😉 I have OBDeleven, my friend has VCP but no idea how to get the file extracted

Can you please share the steps for VCP or do we need other sw?

**Reply by perkelle** [reply]
Try to request file with vctool dataset request service

**Reply by NetCatx** [reply]
Thanks. Does it mean VCP cant do it?

**Reply by jilleb** [reply]
Exactly, VCP can't

**Reply by batal131** [reply]
Unfortunately it's not possible to "read" FPA dataset with any software. 

I received mine with ODIS online.

**Reply by NetCatx** [reply]
That makes it complicated 🤔 I have no access to ODIS

**Comment by deppdive** [comment]
the ODIS WAY ...

**Reply by G1Apples** [reply]
Do i only need ODIS or also an GEKO allowance ? Or is here anyone that could "read"/get the dataset with odis online for me ?
